### PR TITLE
created create project route and edited id column for all project ent…

### LINKED ISSUE #31: [Backend-routes] POST api/v1/projects/
## Description
- This route to add new projects by only the admin. 

### Request body
```
{
  name: string,
  year: number,
}
```
#### Response 201
```
{
message :  تم إضافة المشروع بنجاح 
 } 
```

### Unauthorized users
Fail: 401
```
{
    "statusCode": 401,
    "message": "Unauthorized"
}
```

###  If the user is not the admin
Fail: 403 
```
{
 "statusCode": 403,
 "message": "Forbidden resource",
 "error": "Forbidden"
}
```

Fail: 500
```
{
  message: "حدث خطأ داخلي بالنظام "
  statusCode: 500
}
```

### PR DESCRIPTION
**Pull request by Bas-Shiekh**
…ities.

relates #31

### CODE REVIEW

**Review by MohammedOmar123 — APPROVED** [review]
عاااش ي بسولة

**Review by SaifHayek — APPROVED** [review]

File: server/src/projects/projects.controller.ts [hunk 1]
```diff
@@ -6,14 +6,21 @@ import {
...
   Param,
   Delete,
+  UseGuards,
 } from '@nestjs/common';
 import { ProjectsService } from './projects.service';
 import { CreateProjectDto, UpdateProjectDto } from './dto';
+import { Roles } from 'src/auth/decrator';
+import { Role } from 'src/auth/enums/role.enum';
+import { JwtAuthGuard } from 'src/auth/strategy';
+import { RolesGuard } from 'src/auth/Guards/roles.guard';
```
Comment (MostafaBalousha123) [inline server/src/projects/projects.controller.ts]: good work basola, but fix paths, make it as '../src/auth/decrator'

File: server/src/projects/projects.controller.ts [hunk 2]
```diff
@@ -6,14 +6,21 @@ import {
...
 
 @Controller('projects')
 export class ProjectsController {
   constructor(private readonly projectsService: ProjectsService) {}
 
+  @Roles(Role.Admin)
+  @UseGuards(JwtAuthGuard, RolesGuard)
```
Comment (MostafaBalousha123) [inline server/src/projects/projects.controller.ts:23]: I think first should check if there JWT token and Roles then chack if the role === admin 

- so the code should be in this order

@UseGuards(JwtAuthGuard, RolesGuard)
@Roles(Role.Admin)

File: server/src/projects/projects.service.ts [hunk 3]
```diff
@@ -10,8 +11,13 @@ export class ProjectsService {
...
 
-  create(createProjectDto: CreateProjectDto) {
-    return 'This action adds a new project';
+  async create(dto: CreateProjectDto) {
+    await this.projectRepository.create({
+      name: dto.name,
+      year: dto.year,
+    });
+
+    return CREATE_PROJECT;
```
Comment (MostafaBalousha123) [inline server/src/projects/projects.service.ts]: I think this message should returned as object 

 {  message: CREATE_PROJECT }

**Review by MostafaBalousha123 — CHANGES_REQUESTED** [review]

File: server/src/projects/projects.controller.ts (hunk 2 above)
Comment (MohammedOmar123) [reply server/src/projects/projects.controller.ts:23]: Actually what Basil did here is totally fine, because the "@Roles(Role. Admin)" decorator doesn't validate the request, it just adds some metadata (role in this case) to the controller. And this metadata is required for RolesGuard class.
So RolesGuard will extract the role from the request and compare it to the role in the metadata of the controller and if they are matched the controller will handle the request.
In summary:-  
@roles(Role.Admin) >> will define a role for the controller.
@RolesGuard >> will validate the role in the request to see if it matches the defined role.

**Review by MohammedOmar123 — APPROVED** [review]
